### ISSUE TITLE
Tried the library but no device audio gets recorded

### ISSUE THREAD

**Issue by VincentLu91** [Nilsantos/react-native-screen-audio-recorder#2]
Tried:

I installed and imported the library in my InternalAudio component. However, I set up a ScreenAudioRecorder and it doesn't return any audio data from videos that I play on the device.

Code:
```
...
import ScreenAudioRecorder from 'react-native-screen-audio-recorder';
...
function InternalRecording(props) {
...
  const options = {
    sampleRate: 16000,
      channels: 1,
      bitsPerSample: 16,
      fileName: 'novo.wav',
      fromMic: false,
      saveFile: false,
      audioEmitInterval: 128000,
  };

...
  async function startTranscribing() {
    ScreenAudioRecorder.init(options);
    ....
   global.socket.onmessage = (message) => {
      console.log("Entering onmessage");
      console.log("onsocket message is: ", message);
      let msg = '';
      const res = JSON.parse(message.data);
      texts[res.audio_start] = res.text;
      const keys = Object.keys(texts);
      keys.sort((a, b) => a - b);
      for (const key of keys) {
        if (texts[key]) {
          msg += ` ${texts[key]}`;
        }
      }
      console.log("Leaving onmessage. msg is: ", msg);
      setTranscript(msg);
    };
   ...
    ScreenAudioRecorder.start();
    global.socket.onopen = () => {
      ScreenAudioRecorder.on('data', data => {
        if (global.socket) {
          global.socket.send(JSON.stringify({audio_data: data}));
        }
      })
    }
...
ScreenAudioRecorder.stop();
...
```

Android version is version 10 on my device.

React Native version is 0.63.

**Comment by Nilsantos** [comment]
> ScreenAudioRecorder.on('data', data => {



Are you sure the global.socket is not null?
Did you try putting a log right after "ScreenAudioRecorder.on('data', date => {" to see if something is returned?

**Comment by VincentLu91** [comment]
nothing's returned. First I was prompted to allow casting/recording and I confirmed yes. But the console.log doesn't run.

**Comment by Nilsantos** [comment]
Try somenting like this:

...

    ScreenAudioRecorder.on('data', data => {
        console.log(data);
        if (global.socket) {
            global.socket.send(JSON.stringify({audio_data: data}));
        }
    })

    global.socket.onopen = () => {
       console.log('start');
       ScreenAudioRecorder.start();
    }

...

**Comment by VincentLu91** [comment]
Still nothing

**Comment by VincentLu91** [comment]
Hi @Nilsantos . Any other suggestions or ideas?

**Comment by niembro64** [comment]
This library doesn't work? Been trying it for a few hours now.

(EDIT) I tried the demo and it works just fine! I just need to figure out how I'm using it wrong! 🍔

**Comment by Nilsantos** [comment]
@niembro64 yes, the library works. Could you share the error that appeared? I'm not updating this library anymore, but I'll try to fix the problems.

**Comment by niembro64** [comment]
Hi @Nilsantos how exactly are you accomplishing this? Which android MediaRecorder.AudioSource are you using when it's not recording from the mic? REMOTE_SUBMIX seems to be what's needed but it seems to require special permissions?

https://developer.android.com/reference/android/media/MediaRecorder.AudioSource

I'm making an app that's heavily dependent on this stuff so any insight is helpful!

**Comment by Nilsantos** [comment]
@niembro64, you can see the android source code here https://github.com/Nilsantos/react-native-screen-audio-recorder/tree/master/android/src/main/java/com/reactnativescreenaudiorecorder.

But is an error appearing when you try to use the library in react native?
If you can share what's going on maybe I can fix it

**Comment by niembro64** [comment]
@Nilsantos I appreciate the responses.

To be clear, I have tried your demo app and it works as intended!

I am curious how this is getting accomplished i.e. which audio source number you are using in android for capturing audio output from the phone? 

Here's a snapshot of your Java code:
![image](https://user-images.githubusercontent.com/49417819/214982312-30161f4c-a77e-4ce6-a031-97e07f2cb464.png)

I see you are using VOICE_RECOGNITION (6) when fromMic but I can't understand what audio source is being used when not fromMic?

**Comment by niembro64** [comment]
@Nilsantos I realized what my issue is -> I'm using two libraries that both us the same parent library / underlying java methods.

+ react-native-live-audio-stream ---------------- doing DSP analysis from microphone
+ react-native-screen-audio-recorder ---------- capturing my real-time sound effects produced by phone to .wav file

Some issues noticed when using together:
+ The .on('data') method for one of them seems to overlap or override the other, i.e. I can't get both .on('data') methods to run 
+ The .wav file from your library is produced and has the proper length, but using both libraries simultaneously produces silence for the whole length of the file instead of capturing audio output. 

Appreciate any insight you might have!
Eric
